### PR TITLE
server: support ComChangeUser

### PR DESCRIPTION
**Pull request by coocood**
For compatibility.

### CODE REVIEW

**Comment by winkyao** [comment]
Is this PR ready to be reviewed?

**Comment by coocood** [comment]
@winkyao yes

**Comment by coocood** [comment]
@tiancaiamao PTAL

**Comment by shenli** [comment]
LGTM

File: server/conn.go [hunk 1]
```diff
@@ -1023,3 +1029,26 @@ func (cc *clientConn) upgradeToTLS(tlsConfig *tls.Config) error {
...
+
+func (cc *clientConn) handleChangeUser(data []byte) error {
+	user, data := parseNullTermString(data)
+	cc.user = hack.String(user)
+	if len(data) < 1 {
+		return mysql.ErrMalformPacket
+	}
+	passLen := data[0]
+	data = data[1:]
+	pass := data[:passLen]
```
Comment (tiancaiamao) [inline server/conn.go]: Will malformat packet make TiDB process crash?

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**